SU2_CFD/src/iteration/CFluidIteration.cpp

```diff
@@ -226,12 +226,37 @@ bool CFluidIteration::Monitor(COutput* output, CIntegration**** integration, CGe
 
   UsedTime = StopTime - StartTime;
 
+  unsigned long Iter= config[val_iZone]->GetInnerIter();
+
   if (config[val_iZone]->GetMultizone_Problem() || config[val_iZone]->GetSinglezone_Driver()) {
     output->SetHistory_Output(geometry[val_iZone][INST_0][MESH_0], solver[val_iZone][INST_0][MESH_0], config[val_iZone],
                               config[val_iZone]->GetTimeIter(), config[val_iZone]->GetOuterIter(),
                               config[val_iZone]->GetInnerIter());
   }
 
+  /*--- Turbomachinery Specific Montior ---*/
+  if (config[ZONE_0]->GetBoolTurbomachinery() && config[val_iZone]->GetSinglezone_Driver()){
+
+    /*--- Turbomachinery Performance Computation ---*/
+    if (val_iZone == config[ZONE_0]->GetnZone()-1)
```

Comment:
Why not just ZONE_0 ? You want the last? (but there single zone driver only does one :thinking: )